mucommander-core/src/main/java/com/mucommander/ui/layout/AsyncPanel.java

```diff
@@ -103,19 +104,15 @@ public void ancestorMoved(AncestorEvent event) {}
      * Loads the target component by calling {@link #getTargetComponent()} and replace the wait component by it.
      */
     private void loadTargetComponent() {
-        new Thread() {
-            @Override
-            public void run() {
-                JComponent targetComponent = getTargetComponent();
-
+        new Thread(() -> {
+            JComponent targetComponent = getTargetComponent();
+            SwingUtilities.invokeLater(() -> {
```

Comment:
can't we drop the new thread from here now that we register the code execution in SwingUtilities#invokeLater? the call to getTargetComponent should be a quick one..

Reply:
I would say that we don't want to block Swing threads - we run in a separate thread as much as we can then any UI related things are being done by Swing thread it feels is OK to update UI ("This method should be used when an application thread needs to update the GUI. ").

Reply:
right, but do we know that creating and running a thread is quicker than calling getTargetComponent? I don't think we really care about msec - so if it's comparable (or in other words, if we don't see a clear benefit of initiating another thread), I'd drop the new thread here..

Reply:
> we know that creating and running a thread is quicker than calling getTargetComponent?

For example, https://github.com/mucommander/mucommander/blob/c58c4c7a4d002735227ba92c5681c3e527828f8e/mucommander-core/src/main/java/com/mucommander/ui/viewer/FileFrame.java#L93 - this can be painfully slow operation when opening large file (as text or image).
Spotted that during debugging.

Reply:
okok, you're right - I wrote this comment without looking again at AsyncPanel and I forgot that this is the async part that is supposed to do the heavy lifting (it looked like an innocent method that returns a Swing component)